enforcer/src/main/java/org/wso2/micro/gateway/enforcer/keymgt/KeyManagerHolder.java

```diff
@@ -0,0 +1,159 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.gateway.enforcer.keymgt;
+
+import com.google.gson.Gson;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.gateway.discovery.keymgt.KeyManagerConfig;
+import org.wso2.micro.gateway.enforcer.config.ConfigHolder;
+import org.wso2.micro.gateway.enforcer.config.dto.JWKSConfigurationDTO;
+import org.wso2.micro.gateway.enforcer.config.dto.TokenIssuerDto;
+import org.wso2.micro.gateway.enforcer.constants.APIConstants;
+import org.wso2.micro.gateway.enforcer.discovery.KeyManagerDiscoveryClient;
+import org.wso2.micro.gateway.enforcer.dto.ClaimMappingDto;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * KeyManager holder class for Micro gateway.
+ */
+public class KeyManagerHolder {
+
+    private static final Logger logger = LogManager.getLogger(ConfigHolder.class);
+
+    private static final String X509 = "X.509";
+    private static KeyManagerHolder instance;
+    private Map<String, TokenIssuerDto> tokenIssuerMap = ConfigHolder.getInstance().getConfig().getIssuersMap();
+
+    private KeyManagerHolder() {}
+
+    public static KeyManagerHolder getInstance() {
+        if (instance == null) {
+            instance = new KeyManagerHolder();
+        }
+        return instance;
+    }
+
+    public void init() {
+        KeyManagerDiscoveryClient keyManagerDs =  KeyManagerDiscoveryClient.getInstance();
+        keyManagerDs.watchKeyManagers();
+    }
+
+    public void populateKMIssuerConfiguration(List<KeyManagerConfig> kmIssuers) {
+        Map<String, TokenIssuerDto> resultIssuerList = new HashMap<>();
+        for (KeyManagerConfig keyManagerConfig : kmIssuers) {
+            JSONObject configObj = new JSONObject(keyManagerConfig.getConfiguration());
+            Map<String, Object> configuration = new HashMap<>();
+            Iterator<String> keysItr = configObj.keys();
+            while (keysItr.hasNext()) {
+                String key = keysItr.next();
+                Object value = configObj.get(key);
+                configuration.put(key, value);
+            }
+
+            if (keyManagerConfig.getEnabled()) {
+                updateTokenIssuerList(configuration, resultIssuerList);
+            }
+        }
+        ArrayList<TokenIssuerDto> configIssuerList = ConfigHolder.getInstance().getConfigIssuerList();
```

Comment:
L56 has been already retrieve all the existing issuer list. Can't we use same at here?